### PR TITLE
mpmc_queue modified - issue-114

### PR DESCRIPTION
**Pull request by Drioastorga**
Changes:
- A file with the enum class Queue_modes has been introduced.
- Instead of using a boolean, the queue mode is defined by the aforementioned enum class.
- The constructor with a single parameter has been deleted. 
- Now there is only one contructor with a default value for the second parameter.
- The data members initializations is performed in the initialization list of the constructor.
- The data member that defines the queue mode in the execution models has been modified from bolean type to Queue_mode.

### CODE REVIEW

File: include/common/queue_modes.h [hunk 1]
```diff
@@ -0,0 +1,6 @@
+namespace grppi{
```
Comment (jdgarciauc3m) [inline include/common/queue_modes.h]: Move this to mpmc_queue.h

File: include/common/mpmc_queue.h [hunk 2]
```diff
@@ -28,48 +28,36 @@ namespace grppi{
...
 constexpr int DEFAULT_SIZE = 100;
 
+
+
 template <typename T>
 class mpmc_queue{
    private:
       int size;
       std::vector<T> buffer;
+      Queue_mode lockfree = Queue_mode::blocking;
```
Comment (jdgarciauc3m) [inline include/common/mpmc_queue.h]: naming: queue_mode

Comment (jdgarciauc3m) [reply include/common/mpmc_queue.h]: and the member variable should be mode

**Review by jdgarciauc3m — CHANGES_REQUESTED** [review]
See comments

File: include/common/mpmc_queue.h (hunk 2 above)
Comment (Drioastorga) [reply include/common/mpmc_queue.h]: Queue_mode has been renamed to queue_mode and the member variable has been renamed to mode.

**Review by jdgarciauc3m — APPROVED** [review]
OK

File: include/common/queue_modes.h (hunk 1 above)
Comment (Drioastorga) [reply include/common/queue_modes.h]: The definition of the enum class has been moved to mpmc_queue.h. Aditionally, the include of this header file has been moved form the common header file to the execution model headers files.

**Comment by jdgarciauc3m** [comment]
Solves issue #114

**Pull request merged**